recipes/stlab/all/conanfile.py

```diff
@@ -11,48 +11,169 @@ class Stlab(ConanFile):
     license = 'BSL-1.0'
     topics = 'conan', 'c++', 'concurrency', 'futures', 'channels'
 
-    settings = 'compiler'
+    settings = "arch", "os", "compiler", "build_type", 
 
     options = {
-        'coroutines': [True, False]
+        "boost_optional": [True, False],
+        "boost_variant": [True, False],
+        "coroutines": [True, False],
+        "task_system": ["portable", "libdispatch", "emscripten", "pnacl", "windows", "auto"],
     }
+
     default_options = {
-        'coroutines': False
+        "boost_optional": False,
+        "boost_variant": False,
+        "coroutines": False,
+        "task_system": "auto",
     }
 
     no_copy_source = True
     _source_subfolder = 'source_subfolder'
 
-    requires = 'boost/1.69.0'
+    def _use_boost(self):
+        return self.options.boost_optional or self.options.boost_variant
+
+    def _requires_libdispatch(self):
+        # On macOS it is not necessary to use the libdispatch conan package, because the library is 
+        # included in the OS.
+        return self.options.task_system == "libdispatch" and self.settings.os != "Macos"
+
+    def requirements(self):
+        if self._use_boost():
+            self.requires("boost/1.75.0")
+
+        if self._requires_libdispatch():
+            self.requires("libdispatch/5.3.2")
 
     def source(self):
         tools.get(**self.conan_data["sources"][self.version])
         extracted_dir = "libraries-" + self.version
         os.rename(extracted_dir, self._source_subfolder)
 
-    def configure(self):
+    def _fix_boost_components(self):
+        if self.settings.os != "Macos": return
+        if self.settings.compiler != "apple-clang": return
+        if Version(self.settings.compiler.version) >= "12": return
+
+        #
+        # On Apple we have to force the usage of boost.variant, because Apple's implementation of C++17 is not complete.
+        #
+        self.output.info("Apple-Clang versions less than 12 do not correctly support std::optional or std::variant, so we will use boost::optional and boost::variant instead.")
+        self.options.boost_optional = True
+        self.options.boost_variant = True
```

Comment:
I did some digging and our convention has been to change the default in `config_options`

Reply:
Oops, I left this function by mistake. 
It was replaced by `_validate_boost_components` as you suggested.

Reply:
Please remove this code since it's not used

```suggestion
```